core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -229,12 +225,7 @@ protected boolean resolveIndex() {
         preVersionTypes[requestIndex] = updateRequest.versionType();
         //  We need to do the requested retries plus the initial attempt. We don't do < 1+retry_on_conflict because retry_on_conflict may be Integer.MAX_VALUE
         for (int updateAttemptsCount = 0; updateAttemptsCount <= updateRequest.retryOnConflict(); updateAttemptsCount++) {
-            UpdateResult updateResult;
-            try {
-                updateResult = shardUpdateOperation(metaData, request, updateRequest, indexShard);
-            } catch (Exception t) {
-                updateResult = new UpdateResult(null, null, false, t, null);
-            }
+            UpdateResult updateResult = shardUpdateOperation(metaData, request, updateRequest, indexShard);
             if (updateResult.success()) {
```

Comment:
since this one already uses the `.success()` method, maybe do the same on WriteResult?